sphinx/ext/autodoc.py

```diff
@@ -514,18 +515,22 @@ def process_doc(self, docstrings):
             for line in docstringlines:
                 yield line
 
-    def add_content(self, more_content, no_docstring=False):
-        """Add content from docstrings, attribute documentation and user."""
-        # set sourcename and add content from attribute documentation
+    def get_sourcename(self):
         if self.analyzer:
             # prevent encoding errors when the file name is non-ASCII
-            if not isinstance(self.analyzer.srcname, text_type):
-                filename = text_type(self.analyzer.srcname,
-                                     sys.getfilesystemencoding(), 'replace')
+            if not isinstance(self.analyzer.srcname, unicode):
+                filename = unicode(self.analyzer.srcname,
+                                   sys.getfilesystemencoding(), 'replace')
```

Comment:
This 3 lines change is a cause of test failing on Py3. `text_type` is a compatibility type for py2 `unicode` and py3 `str`. please revert this 3 lines.